packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -363,6 +374,40 @@ contract OPContractsManager is ISemver {
         return output;
     }
 
+    /// @notice Upgrades a set of chains to the latest implementation contracts
+    /// @param _systemConfigs Array of SystemConfig contracts, one per chain to upgrade
+    /// @param _proxyAdmins Array of ProxyAdmin contracts, one per chain to upgrade
+    /// @dev This function is intended to be called via DELEGATECALL from the Upgrade Controller Safe
+    function upgrade(ISystemConfig[] calldata _systemConfigs, IProxyAdmin[] calldata _proxyAdmins) external {
```

Comment:
1. let's combine these args into an array of structs.
2. Also missing: upgrading the SuperchainConfig and ProtocolVersions
3. check that all chains have the same SuperchainConfig (sanity check this assumption with @sbvegan).

Reply:
add todos